src/AdminConsole/Pages/Initialize.cshtml.cs

```diff
@@ -0,0 +1,53 @@
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using Passwordless.AdminConsole.Db;
+using Passwordless.AdminConsole.Services.PasswordlessManagement;
+using Passwordless.Common.Models.Apps;
+
+namespace Passwordless.AdminConsole.Pages;
+
+public class Initialize(ConsoleDbContext dbContext, IPasswordlessManagementClient managementClient) : PageModel
+{
+    public string ManagementKey { get; private set; }
+    public string AdminEmail { get; private set; }
+    public string ApiUrl { get; private set; }
+
+    public CreateAppResultDto CreateAppResultDto { get; private set; }
+
+    public async Task<IActionResult> OnGetAsync()
+    {
+        if (!await dbContext.Database.CanConnectAsync())
+        {
+            return Page();
+        }
+
+        return RedirectToPage("/");
+    }
+
+    public void OnGetAsync(CreateAppResultDto keys, string managementKey, string apiUrl)
+    {
+        CreateAppResultDto = keys;
+        ManagementKey = managementKey;
+        ApiUrl = apiUrl;
+    }
+
+    public async Task OnPostAsync()
```

Comment:
You might want to use `Blazor` for this, so you can use the form confirmation dialog.

Also generate a random `appid`. This creates an enumeration vulnerability.

Reply:
Any specific reason to use form confirmation? I think we're good without it.

And I don't see any enumeration problem with this?

Reply:
There's not an enumeration problem. However, someone could post to the endpoint and spam create apps. I can put a check in there to make sure the app doesn't exist already.

Reply:
I don't know, could avoid some paranoia for targeted attacks against specific `appid`

Reply:
I converted this over to blazor...for the practice of using blazor. We don't accept an appId so they can't really target an attack against that as far as I'm aware of.